functions/go/set-image/image_transformer.go

```diff
@@ -114,62 +109,72 @@ func (si *SetImage) validateInput() error {
 }
 
 // Config initializes SetImage from a functionConfig sdk.KubeObject
-func (si *SetImage) Config(functionConfig *sdk.KubeObject) error {
+func (si *SetImage) Config(functionConfig *fn.KubeObject) (error, bool) {
 	si.Image = types.Image{}
 	si.AdditionalImageFields = nil
 	switch {
-	case validGVK(functionConfig, "v1", "ConfigMap"):
-		if found, err := functionConfig.Get(&si.Image, "data"); err != nil {
-			return fmt.Errorf("unable to convert functionConfig to v1 ConfigMap:\n%w", err)
-		} else if !found {
-			return fmt.Errorf("unable to get field data from functionConfig")
-		}
-	case validGVK(functionConfig, fnConfigAPIVersion, fnConfigKind):
-		if err := functionConfig.As(si); err != nil {
-			return fmt.Errorf("unable to convert functionConfig to %s %s:\n%w",
-				fnConfigAPIVersion, fnConfigKind, err)
-		}
+	case functionConfig.IsGVK( "v1", "ConfigMap"):
+		functionConfig.GetOrDie(&si.Image, "data")
+	case functionConfig.IsGVK(fnConfigAPIVersion, fnConfigKind):
+		functionConfig.AsOrDie(si)
```

Comment:
What is the convention for when to panic vs return an error?

Reply:
SDK processor handles specific panics like mismatch field type (what `XXXOrDie` raises). This is to give user a better experience. 
Back to your question about when to panic vs return an error:
If a SDK function returns an error, we expect users to handle the error; if a SDK function does not return an error, users should *not* expect panic **raised directly from the SDK side**.

Reply:
The second use case in the description is from the `AsOrDie` panic
```bash
[error]: Resource(apiVersion=fn.kpt.dev/v1alpha1, kind=SetImage, Name=my-func-config) has unmatched field type: 
  Exit code: 1
```